### DISCUSSION TITLE
Kanban view of selected map nodes

### DISCUSSION THREAD

**Discussion by JJSlote** [freeplane/freeplane#1864] [Show and tell]
Greetings - I’ve made an AutoHotkey script to display a Kanban view of selected Freeplane map nodes. The user selects one or more nodes on the map, presses Ctrl-C to execute Freeplane’s Copy Node (branch), and clicks on a UI button in the AHK script. And nodes from a map like this:

![CabbageMap](https://github.com/freeplane/freeplane/assets/68667676/1b66abb3-80d9-44a9-b7f8-5cac4156637a)

… are transformed into a Kanban display like this:

![CabbageKanban](https://github.com/freeplane/freeplane/assets/68667676/7b724dab-399c-4254-8249-a6ee89739b2c)

I think Kanban view is a great capability for Freeplane, and I’ll be attaching a simplified version of the AHK script, primarily as proof of concept. But I’m really hoping someone in Freeplane’s extraordinary scripting community will be inspired to implement a similar filter as a Groovy script that any Freeplane user can deploy and customize. That script would loop through the selected nodes and their fields directly, without the convolutions and ad hockery required to construct Kanban cards from the HTML clipboard.

The Kanban board is obviously compact and information-rich. Cards are variable length and can be rearranged and edited. Columns can be scrolled independently, rearranged, collapsed, and dragged to other Kanban boards in a split screen. _Edited to add:_ The board's appearance is enhanced by a theme and by CSS snippets that can be script-generated and applied on the fly; thus for example <s>toggling the checkbox</s> clicking a link on a branch node card can be set to collapse the cards of its children. 

The board is displayed via Matthew Meyers’ great Kanban plugin:

https://github.com/mgmeyers/obsidian-kanban

… for Obsidian: https://obsidian.md/ . (You’ll find Kanban among Obsidian’s Community Plugins.)

The generated board is just a Markdown file with some YAML frontmatter. Two hashmarks “##” to represent a column, and a column-one hyphen, followed by a space and some text, to identify each card. Fortunately, most HTML tags will be recognized, for formatting and links. The script reads in a prototype file, KanbanProto.md and merges it with the tweaked HTML clipboard to create ScriptGenerated.md , which Obsidian should recognize – and the plugin display – as a Kanban board.

The simplified version of the script, attached in the zip file below, distributes the cards evenly by count across the first four columns, with the remaining cards in the last column. The production version makes a column for every branch node, then combines columns, or splits them, to achieve the desired column count. In the production version, I’ve also added, for most cards, a _freeplane_: URI at the “⇐” symbol to link back to the corresponding node on the map.

Again, this is proof of concept and, I hope, inspiration for a future Groovy script. For anyone with the unnatural desire to try and get the attached AHK script running on their own system: it’s written for AHK version 1.1.36.02; AHK has since moved on to Version 2, with major differences in syntax. It requires the WinClip library/class (easily found and downloaded) to copy the HTML clipboard into a plaintext field with HTML markup. It needs the AHK preamble and a UI, some hotkey or button to activate it. And filepaths and URI’s set to whichever drives, folders and filenames you’re using.

Within the attached zip file you’ll find:

_HTMLClipToKanban.txt_ - The simplified script as a text file
_Cabbage Demo.mm_ - A sample map, Cabbage Demo, with snippets from the Wikipedia entry for Cabbage.
_KanbanProto.md_ - The blank prototype Kanban/Markdown file, the template into which the script inserts its output.
_CabbageMapKanbanSimple.md_ - A Kanban file generated from the simple version of the script, run against selected nodes in Cabbage Demo .
_CabbageMapKanbanFull.md_ - A Kanban file generated from the production version. The _freeplane:_ backlinks have ID_ values matching those in Cabbage Demo but getting the backlinks to work will require text-editing the dummy path to match Cabbage Demo’s install path.

Make of this what you will, and please reach out with any questions!

Cheers…Jerome

[HTMLClipToKanban-FreeplaneDiscussions.zip](https://github.com/user-attachments/files/15753314/HTMLClipToKanban-FreeplaneDiscussions.zip)

**Comment by euu2021** [comment]
Interesting!

At first, I did not see the difference between the Kanban view and a horizontal distribution of the nodes, like shown at https://sourceforge.net/p/freeplane/discussion/758437/thread/524251a7cd/?limit=25#6826. 

But, now that you mentioned `Columns can be scrolled independently, rearranged, collapsed, and dragged to other Kanban boards in a split screen`, I see that it's a different beast. It's more like split tabs with a Jump In in each one:

![javaw_dhheiLkBUd](https://github.com/freeplane/freeplane/assets/77707706/9af76afa-83ba-45cb-a371-ee55cbee9dae)

The problem is that the nodes don't get adjusted to the width of each tab, so it's painful to adjust the Kanban columns manually.

**Reply by JJSlote** [reply]
Right, there's a lot more to Kanban than a view of the origin app in multiple columns. A Kanban user has granular control of the cards, their selection, alignment and juxtaposition; they're like index cards on a table, curated from a larger set. Aside from their traditional use in project management, they're compact and readable, to inform one's writing on a split screen. Just a valuable export/report form for Freeplane data.   

![QKanban3A](https://github.com/freeplane/freeplane/assets/68667676/e3f6c8a9-cbd2-444a-ba7e-7fbfa9cf4a64)

Cheers...Jerome
